lib/src/local_working_copy.rs

```diff
@@ -1967,21 +1992,45 @@ impl TreeState {
                 // the state type as FileType::GitSubmodule if it was before.
                 continue;
             }
-
             // Create parent directories no matter if after.is_present(). This
             // ensures that the path never traverses symlinks.
             let Some(disk_path) = create_parent_dirs(&self.working_copy_path, &path)? else {
                 changed_file_states.push((path, FileState::placeholder()));
                 stats.skipped_files += 1;
                 continue;
             };
+
             // If the path was present, check reserved path first and delete it.
-            let present_file_deleted = before.is_present() && remove_old_file(&disk_path)?;
+            let present_file_deleted = before.is_present()
+                && if matches!(before.as_normal(), Some(TreeValue::GitSubmodule(_))) {
+                    remove_old_submodule_dir(&disk_path)?
+                } else {
+                    remove_old_file(&disk_path)?
+                };
+
             // If not, create temporary file to test the path validity.
             if !present_file_deleted && !can_create_new_file(&disk_path)? {
-                changed_file_states.push((path, FileState::placeholder()));
-                stats.skipped_files += 1;
-                continue;
+                /////// This should only print if files will be found as added --- so check if the submodule path is ignored? How? Can UI print it instead?
+                if matches!(before.as_normal(), Some(TreeValue::GitSubmodule(_))) {
+                    eprintln!(
+                        "Files from the submodule will appear as added. Add the submodule to `.git/info/exclude` and run `jj file untrack` to ignore the directory contents."
+                    );
+                }
```

Comment:
This `if` needs correct logic to check if the path is ignored. Or we just drop the print.